src/index.test.ts

```diff
@@ -104,3 +105,75 @@ test('Test placeholder', async (t) => {
     user_id,
   });
 });
+
+test("Aggregator correctly forms cache query", async (t) => {
+  const readStub = stub();
+
+  readStub
+    .withArgs(
+      "select * from aggregate_cache where id = $1",
+      ['b83753e3336aaa7544d02abf12e085a3c95b96b2916acc96b36d5f5652f21723'],
+    )
+    .resolves(
+      fakePoolResult([
+         {
+           id: 'uuid',
+           aggregate_type: 'NOT NULL',
+           data: {some: 'stuff'},
+           time: 'iso time',
+         },
+        ],
+      ),
+    );
+
+  readStub
+    .resolves(
+      fakePoolResult(),
+    );
+  // this part should execute exactly once - there should only be one occurance of
+  // this in readStub.args
+
+  const store = await newEventStore(getFakePool(readStub), emit);
+
+  const base_query =
+    "SELECT * FROM events WHERE data->>'user_id' = $1 ORDER BY time ASC";
+
+  const testAggregate = store.registerAggregate<[string], Option<User>>(
+    'RandoAggregate',
+    base_query,
+    None,
+    [
+      [
+        isAccountCreated,
+        (acc: any, d: any) => Some({
+          ...acc.getOrElse({}),
+          user_id: d.user_id,
+          name: d.name,
+          email: d.email,
+        }),
+      ],
+      [ isNameChanged, (acc: any, d: any) => Some({ ...acc.getOrElse({}), name: d.name }) ],
+      [ isEmailChanged, (acc: any, d: any) => Some({ ...acc.getOrElse({}), email: d.email }) ],
+    ],
+  );
+
+  const result = await testAggregate(user_id);
+
+  console.log("READSTUB ARGS");
+  console.debug(readStub.args);
+
+  const target_query =
+    'SELECT * FROM (SELECT * FROM events WHERE data->>\'user_id\' = $1' +
+    ' ORDER BY time ASC) as events where events.time > (iso time) order by events.time asc;';
```

Comment:
If you make this a template string, you should be able to get rid of the `.replace(...)` stuff in your assertion. You can also then insert `base_query` into it.